linkerd/app/src/env.rs

```diff
@@ -500,6 +520,51 @@ pub fn parse_config<S: Strings>(strings: &S) -> Result<super::Config, EnvError>
     })
 }
 
+fn oc_trace_labels(oc_file_path: Result<Option<String>, EnvError>) -> HashMap<String, String> {
```

Comment:
I find it somewhat strange that we pass the path in as a `Result`, rather than just not calling this function if we can't get it from the env. What if instead of matching that result here, we made this
```rust
fn oc_trace_labels(oc_file_path: &str) -> HashMap<String, String> {
    // ...
}
```
removed a lot of the error handling logic from this function, and changed the call site to something like
```rust
let mut labels = oc_labels_file_path
    .map(|path| match path {
        Some(path) => oc_trace_labels(path),
        None => {
            warn!("env variable {} not present", ENV_LABELS_FILE_PATH);
            HashMap::new()
        }
    }).unwrap_or_else(|err| {
        warn!(
            "could not read env variable {}: {}",
            ENV_LABELS_FILE_PATH, err
        );
        HashMap::new()
});
``` 
or something like that?

This function feels like it's doing a lot of error handling.

Reply:
Yep, Definitely make sense :+1: